package NEmbeddingsTransfer.NProto;

enum ECompressionType {
    NO_COMPRESSION = 0;
    AUTO_MAX_COORD_RENORM_TO_UI8 = 1;
    AUTO_MAX_COORD_RENORM_TO_UI8_WITH_SCALE = 2;
}

message TCompressionInfo {
    optional float Scale = 1 [default = 1.0];
}

message TModelEmbedding {
    optional string Name = 1;
    optional string Version = 2;
    optional bytes CompressedData = 3;
    optional ECompressionType CompressionType = 4;
    optional TCompressionInfo CompressionInfo = 5;
}
